SU2_CFD/src/output/filewriter/CCGNSFileWriter.cpp

```diff
@@ -0,0 +1,305 @@
+    BCType_t bocotype;
+    DataType_t datatype;
+    GridLocation_t location;
+    ElementType_t type;
+    DataClass_t dataclass;
+
+    unsigned short iDim = 0, nDim = dataSorter->GetnDim();
+    unsigned short varStart = 2;
+    if (nDim == 3) varStart++;
+    char boconame;
+    char basename[] = "Base";
+    char zonename[] = "Zone 1";
+    char coordname[32];
+    char ElementSectionName[32];
+    unsigned long jVar;
+    unsigned short NVar, iVar, iPoint;
+    stringstream name, secname, ss;
+    string mesh_file, base_file, wall_string, far_string, null_string, str;
+    unsigned long nLocalVertex   = dataSorter->GetnPoints();
+    nElem = dataSorter->GetnElemGlobal();
+
+    isize[0][0] = (cgsize_t)nPoint_Global;        // vertex size
+    isize[1][0] = (cgsize_t)nElem;                // cell size
+    isize[2][0] = 0;                              //(cgsize_t)nSurf_Elem; boundary vertex size (zero if elements not sorted)
+
+    /*---    Open Mesh File    ---*/
+    mesh_file = config->GetMesh_FileName();
+    cgns_err = cg_open(mesh_file.c_str(), CG_MODE_READ, &cgns_file);
+    if(cgns_err) cg_error_print();
+    nbases = 1;
+    nzones = 1;
+
+    /*---      Create Solution File    ---*/
+    base_file = fileName;
+    cgns_err = cg_open(base_file.c_str(), CG_MODE_WRITE, &cgns_file1);
+    if(cgns_err) cg_error_print;
+    cell_dim = 3;
+    phys_dim = 3;
+    cgns_err = cg_base_write(cgns_file1, basename, cell_dim, phys_dim,  &base_number);
+    if(cgns_err) cg_error_print;
+    cgns_err = cg_zone_write(cgns_file1, base_number, zonename, *isize, Unstructured, &zone_number);
+
+    /*---    Write Coordinates    ---*/
+    cgns_err = cg_ncoords(cgns_file, nbases, nzones, &ncoords);
+    if(cgns_err) cg_error_print();
+    for (i = 1; i < ncoords + 1; i++){
+        cgns_err = cg_coord_info(cgns_file, nbases, nzones, i, &datatype, coordname);
+        if(cgns_err) cg_error_print();
+        start = 1;
+        end = isize[0][0];
+        vector<double> buf(end);
+        cgns_err = cg_coord_read(cgns_file, nbases, nzones, coordname,
+                                 datatype, &start, &end, buf.data());
+        if(cgns_err) cg_error_print();
+        cgns_err = cg_coord_write(cgns_file1, base_number, zone_number, datatype, coordname,
+                                  buf.data(), &coord_index);
+        if(cgns_err) cg_error_print();
+    }
+
+    /*---    Write Connectivity    ---*/
+    cgns_err = cg_nsections(cgns_file, nbases, nzones, &nsections);
+    if(cgns_err) cg_error_print();
+    for ( i = 1; i < nsections + 1; i++)    {
+        cgns_err = cg_section_read(cgns_file, nbases, nzones, i,
+                                   ElementSectionName, &type, &start,
+                                   &end, &nbndry, &parent_flag);
+        if(cgns_err) cg_error_print();
+        cgns_err = cg_ElementDataSize(cgns_file, nbases, nzones, i, &ElementDataSize);
+        if(cgns_err) cg_error_print();
+        if(type != 20){
```

Comment:
use named constant to keep the code easy to read instead of hard coded value
20 ==> 
#define MIXED 20
if (type != MIXED)

Reply:
MIXED defined, now it is more easy to understand.